### PR TITLE
Improved authentication

### PR DESCRIPTION
**Pull request by Malak67**
# Generic Pull Request template

## Linked issue

Please link the issue:

ex. Closes #<33>.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.
- [x] The changes impact front-end.

## For changes impacting frontend

Please leave some screenshots/video recordings of the newly implemented interface and the Figma link for its initial design.

### Screenshots/ recordings

=> Multimedia here.
image

### Figma link

=> Figma link here.

## Description

Please provide a short description about what the PR contains.

## Funny gif

Leave a funny gif for faster review.

### CODE REVIEW

File: services/client/src/api/pools/useGetJoinedPools.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@ import { API } from '../../constants';
 import { useAuth } from '../../hooks';
 import { HypePool } from '../../models';
 
-const getJoinedPools = (address: string) => {
-  const url = `${API}/pools/joined/${address}`;
+const getJoinedPools = () => {
```
Comment (VargaElod23) [inline services/client/src/api/pools/useGetJoinedPools.ts:7]:
```suggestion
const getJointPools = () => {
```
In every place 😄

File: services/client/src/pages/Profile/Profile.effects.tsx [hunk 2]
```diff
@@ -21,15 +21,15 @@ export const useProfileEffects = () => {
   const [joinedPools, setJoinedPools] = useState<HypePool[]>([]);
   const [createdPools, setCreatedPools] = useState<HypePool[]>([]);
   const [currentRewardsNo, setCurrentRewardsNo] = useState<number>(null);
-  const { data } = useGetMyRewards(account);
-  const { data: fetchedJoinedPolls } = useGetJoinedPools(account);
+  const { data } = useGetMyRewards();
+  const { data: fetchedJoinedPolls } = useGetJoinedPools();
```
Comment (VargaElod23) [inline services/client/src/pages/Profile/Profile.effects.tsx:25]:
```suggestion
  const { data: fetchedJointPools } = useGetJoinedPools();
```

File: services/hype-backend/config/auth.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ import { registerAs } from '@nestjs/config';
 
 export default registerAs('auth', () => ({
   walletEndpoint: process.env.WALLET_HOST || 'http://localhost:3006',
+  secret: process.env.AUTH_SECRET || 'secret',
+  tokenExpiry: process.env.JWT_EXPIRY_TIME || '30 days',
```
Comment (VargaElod23) [inline services/hype-backend/config/auth.ts:6]:
```suggestion
  tokenExpiry: process.env.JWT_EXPIRY_TIME || '7 days',
```
The default value in the .env.example is set to 7, let's keep it consistent here too.

**Review by VargaElod23 — APPROVED** [review]
Some small typos and consistency suggestions added but overall LGTM. 👍🏼 for the e2e

File: services/client/src/api/pools/useGetJoinedPools.ts (hunk 1 above)
Comment (rattrap) [reply services/client/src/api/pools/useGetJoinedPools.ts:7]: I don't think that's correct.

File: services/hype-backend/src/modules/auth/jwt-payload.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export interface JWTPayload {
+  address: string;
+  nonce: number;
+}
+
+export interface JWTResponse {
+  accessToken: string;
+}
```
Comment (rattrap) [inline services/hype-backend/src/modules/auth/jwt-payload.ts:8]: Don't need this.

File: services/hype-backend/src/modules/auth/auth.module.ts [hunk 5]
```diff
@@ -1,13 +1,27 @@
...
-  providers: [AuthService],
+  imports: [
+    ConfigModule.forFeature(auth),
+    HttpModule,
+    CqrsModule,
+    JwtModule.registerAsync({
+      imports: [ConfigModule],
+      useFactory: (configService: ConfigService) => ({
+        secret: configService.get('auth.secret'),
+        signOptions: { expiresIn: `${configService.get('auth.tokenExpiry')}` },
```
Comment (rattrap) [inline services/hype-backend/src/modules/auth/auth.module.ts:19]: Don't need the expiry as we're not creating tokens on this side.

File: services/hype-backend/config/auth.ts [hunk 6]
```diff
@@ -2,6 +2,8 @@ import { registerAs } from '@nestjs/config';
 
 export default registerAs('auth', () => ({
   walletEndpoint: process.env.WALLET_HOST || 'http://localhost:3006',
+  secret: process.env.AUTH_SECRET || 'secret',
+  tokenExpiry: process.env.JWT_EXPIRY_TIME || '7 days',
```
Comment (rattrap) [inline services/hype-backend/config/auth.ts:6]: Don't need the expiry as we're not creating tokens on this side.

File: services/hype-backend/.env.example [hunk 7]
```diff
@@ -12,6 +12,10 @@ DB_SSL_REJECT_UNAUTHORIZED=false
 
 # --- AUTH ---
 WALLET_HOST=http://localhost:3006
+# Secret phrase used to encrypt JWT tokens
+AUTH_SECRET=thisisnotasecret
+# Token expiration time
+JWT_EXPIRY_TIME="7 days"
```
Comment (rattrap) [inline services/hype-backend/.env.example:18]: Don't need the expiry as we're not creating tokens on this side.

File: services/client/src/pages/Profile/Profile.effects.tsx [hunk 8]
```diff
@@ -45,7 +45,7 @@ export const useProfileEffects = () => {
   };
 
   useEffect(() => {
-    if (account && data) {
+    if (account && data?.totalUnclaimed) {
       setCurrentRewardsNo(data.totalUnclaimed.length);
```
Comment (rattrap) [inline services/client/src/pages/Profile/Profile.effects.tsx:49]: We don't need this currentRewards thing at all

**Review by rattrap — CHANGES_REQUESTED** [review]

File: services/client/src/pages/Profile/Profile.effects.tsx (hunk 8 above)
Comment (Malak67) [reply services/client/src/pages/Profile/Profile.effects.tsx:49]: We use it in the RewardsContainer. Maybe with the next frontend updates we might not need it but currently it is used.

Comment (rattrap) [reply services/client/src/pages/Profile/Profile.effects.tsx:49]: I mean we don't need to keep it in state as we can get the values directly from `data.totalUnclaimed.length`

**Review by rattrap — CHANGES_REQUESTED** [review]

**Pull request merged**